frappe/desk/doctype/dashboard_chart/dashboard_chart.py

```diff
@@ -86,23 +86,32 @@ def get_chart_config(chart, filters, timespan, timegrain, from_date, to_date):
 	return chart_config
 
 
-def get_group_by_chart_config(chart, filters):
+def get_group_by_chart_config(chart, filters, timespan, from_date, to_date):
+	if not to_date:
+		to_date = datetime.datetime.now()
+
+	if not from_date:
+		from_date = get_from_date_from_timespan(to_date, timespan)
```

Comment:
@surajshetty3416 Let's move this outside the function to avoid code duplication.